### ISSUE TITLE
vxlan: Broken network connectivity when host returns after a reboot

### ISSUE THREAD

**Issue by dthornley** [rancher/rancher#8698]
**Rancher Versions:**
Server: v1.5.9
healthcheck: v0.2.3
vxlan: v0.0.7
network-services: v0.1.2
scheduler: v0.5.1

**Docker Version:**
1.12.6

**OS and where are the hosts located? (cloud, bare metal, etc):**
bare metal running just debian 8 and docker. Around 30 physical hosts in the network.

**Setup Details: (single node rancher vs. HA rancher, internal DB vs. external DB)**
single node rancher, internal DB

**Environment Type: (Cattle/Kubernetes/Swarm/Mesos)**
Cattle

**Steps to Reproduce:**
Reboot a host in the network. 

**Results:** 
The containers on this host will no longer be able to connect to other hosts. Seen when nodes have been down/out of contact for less than 5 minutes.

I never saw this running v1.3.x, but I've been seeing this more often in v1.5. Currently I'm running v1.5.9 and if a physical host is rebooted the containers seem to become isolated from the rest of the network.  DNS seems to work as ping <name> resolves to the correct IP address of the container, but the actual ping fails with 'Destination Host Unreachable'

I've removed the host from the network and then added it again, but even after doing an rm -rf /var/lib/docker first the containers on the host are unable to connect to other containers.

I've seen there is a similar issue #8688 but this is using IPSEC and not the vxlan layer and just restarting the vxlan containers on the host that was rebooted has no effect.

This weekend I even did a complete reinstall of rancher as I'd been upgrading version by version from v1.2, but this didn't help.

**Comment by aemneina** [comment]
@dthornley does your agent host IP change whenever its rebooted? Does the agent host IP match what your hosts IP is?

**Comment by dthornley** [comment]
The hosts IPs don't change as they are using fixed IP addresses. According to the hosts page in the web interface the rancher-agent container doesn't have an IP or do you mean something else?

**Comment by boshido** [comment]
I got this error too 

Rancher Versions:
Server: v1.5.9
Healthcheck : v0.2.3 
Network-services: v0.1.2
Scheduler: v0.6.3

Docker Version:
 17.03.0-ce

OS and where are the hosts located? (cloud, bare metal, etc):
CentOS Linux 7 (with KVM) (3.10.0)

Setup Details: (single node rancher vs. HA rancher, internal DB vs. external DB)
single node rancher, RDS DB

Environment Type: (Cattle/Kubernetes/Swarm/Mesos)
Cattle

Steps to Reproduce:
Reboot a host in the network.

Results:
After finish rebooting the services change itself state from red to be green 
Everything look fine but when I try connect to the service the browser response
```
This site can’t be reached
http://URL/system/health is unreachable.
``` 

How to fix:
Manually restart a load balancer will solve this problem.

**Comment by dthornley** [comment]
I've seen that if I delete the system stacks (healthcheck, scheduler, vxlan and the network-services) for the environment and then add them again the rebooted/restarted nodes are then able to communicate with the network.
This is only a workaround as I have to bring the application down to be able to do this.

**Comment by johnhoran** [comment]
### Rancher Versions:
Server: v1.6.5
healthcheck: v0.3.1
vxlan: v0.11.3
network-services: v0.7.4
scheduler: v0.8.2

### Docker Version:
17.3.2-ce

### OS and where are the hosts located? (cloud, bare metal, etc):
bare metal running ubuntu 16.04.2 with KVM. 
### Setup Details: (single node rancher vs. HA rancher, internal DB vs. external DB)
single node rancher, internal DB

### Environment Type: (Cattle/Kubernetes/Swarm/Mesos)
Cattle
Each host runs a single service container

### Steps to Reproduce:
Reboot a host in the network.

### Results:
I'm able to ping containers on other hosts from the vxlan container.  But I'm not able to ping any other from my service container.  If I deploy another container on the same host, then the networking between the new container and the bad one works correctly.  Restarting the host, or the container don't fix the issue, my work around for now is to delete the container and then recreate it.

Here is the vxlan config:
`
version: '2'
services:
  vxlan:
    network_driver: &id001
      cniConfig:
        10-rancher-vxlan.conf:
          bridge: docker0
          bridgeSubnet: 10.42.0.0/16
          hairpinMode: false
          hostNat: true
          ipam:
            isDebugLevel: 'false'
            logToFile: /var/log/rancher-cni.log
            routes:
            - dst: 169.254.169.250/32
            type: rancher-cni-ipam
          isDebugLevel: 'false'
          isDefaultGateway: true
          linkMTUOverhead: 50
          logToFile: /var/log/rancher-cni.log
          mtu: 1500
          name: rancher-cni-network
          promiscMode: true
          type: rancher-bridge
      defaultNetwork:
        dns:
        - 169.254.169.250
        dnsSearch:
        - rancher.internal
        hostPorts: true
        name: vxlan
        subnets:
        - networkAddress: 10.42.0.0/16
          cidrSize: 24
        defaultPolicyAction: allow
      name: Rancher VXLAN
    start_on_create: true
  cni-driver:
    network_driver: *id001
    scale: 1
    start_on_create: true
`

`
version: '2'
services:
  vxlan:
    cap_add:
    - NET_ADMIN
    image: rancher/net:v0.11.3
    environment:
      RANCHER_DEBUG: 'false'
    network_mode: vxlan
    logging:
      driver: json-file
      options:
        max-file: '2'
        max-size: 25m
    ports:
    - 4789:4789/udp
    command:
    - start-vxlan.sh
    labels:
      io.rancher.container.agent_service.vxlan: 'true'
      io.rancher.sidekicks: cni-driver
      io.rancher.cni.link_mtu_overhead: '0'
      io.rancher.service.selector.link: io.rancher.internal.service.vxlan=true
      io.rancher.network.arpsync: 'true'
      io.rancher.container.create_agent: 'true'
      io.rancher.network.macsync: 'true'
      io.rancher.scheduler.global: 'true'
      io.rancher.internal.service.vxlan: 'true'
  cni-driver:
    privileged: true
    image: rancher/net:v0.11.3
    network_mode: host
    logging:
      driver: json-file
      options:
        max-file: '2'
        max-size: 25m
    pid: host
    command:
    - sh
    - -c
    - touch /var/log/rancher-cni.log && exec tail ---disable-inotify -F /var/log/rancher-cni.log
    labels:
      io.rancher.network.cni.binary: rancher-bridge
      io.rancher.container.dns: 'true'
      io.rancher.service.selector.link: io.rancher.internal.service.vxlan=true
`

**Comment by johnhoran** [comment]
I experimented a bit with the different options in the vxlan, and it seems disabling the "host ports" option in the vxlan config fixes the issue, at least for me.  I'm not exactly sure what that even does, as I haven't noticed any issues from disabling it.

**Comment by sliard** [comment]
Same problem :/
Nobody use rancher in production ?

**Comment by johnhoran** [comment]
I haven't seen it happening since upgrading to v1.6.7.  There were a lot of changes to vxlan in v1.6.5, and I read that upon upgrading you needed to recreate the environment in order to use the new vxlan template.  Anyway I did that upon upgrade, and I haven't seen an issue since.

**Comment by sliard** [comment]
Thx @johnhoran 
No problem since I migrate from 1.6.0 to 1.6.7

**Comment by boshido** [comment]
My problem was already gone also I'm using 1.6.7

**Comment by leodotcloud** [comment]
@dthornley In July 2017, we did a complete refactor of vxlan and that's available in `vxlan:v0.2.0` with image: `rancher/net:v0.11.5`. Since theses changes were not compatible with the older versions, we couldn't allow for an upgrade of the vxlan stack even though rancher/server has been upgraded. Migrating to a new environment is the only way to get these changes.

BTW, very soon we are planning for another release of vxlan which will improve the performance very significantly and unfortunately this might not be an upgradeable option but would be available in a new environment.

**Comment by dthornley** [comment]
@leodotcloud Thanks. Looking forward to the further performance improvements.
